### DISCUSSION TITLE
PathVariable과 QueryParam의 차이와 주된 사용처에 대해 설명해주세요.

### DISCUSSION THREAD

**Discussion by nayonsoso** [2024-woowacourse-study/level-interview#136] [Network]
.

**Comment by Kimprodp** [comment]
**[PathVariable (경로 변수)]**
- 고유하게 리소스를 식별할 때 사용된다. 예를 들어 특정 사용자나 상품을 조회할 때 경로 변수를 지정하여 리소스를 식별할 수 있다.
- 리소스의 기본적인 CRUD 기능에 사용된다.
- 예시
```
/users/{userId} [GET] // 사용자 정보 조회
/users/{userId} [POST] // 사용자 생성
/users/{userId} [PUT] # 사용자 수정
/users/{userId} [DELETE] # 사용자를 삭제
```
**[Query Parameter (쿼리 파라미터)]**
- 필터링,  검색 조건, 페이징과 같이 선택적 정보를 제공할 때 사용된다.
- 예시
```
/users?name=John [GET] // 사용자 이름으로 검색
/products?category=food [GET] // 특정 조건으로 필터링
/items?page=2&size=20 [GET] // 페이징
```

**Comment by seunghye218** [comment]
## PathVariable

**PathVariable**은 URL 경로의 일부로 전달되는 변수를 의미합니다. 이는 주로 RESTful API 설계에서 리소스를 식별하기 위해 사용됩니다.

- 특징
  - **경로 기반**: URL의 일부로 포함되기 때문에 경로 매개변수로 사용됩니다.
  - **식별 목적**: 리소스를 식별하거나 특정 리소스에 접근할 때 사용됩니다.

- 주 사용처
  - 특정 리소스를 식별하고자 할 때: 예를 들어, 사용자의 ID, 상품의 ID, 주문의 ID 등을 URL 경로에서 직접 제공할 때 사용됩니다.
  - RESTful API 설계에서 자원(Resource)을 나타내는 경로를 구성할 때.

### QueryParam

**QueryParam**(또는 RequestParam)은 URL의 쿼리 문자열에서 전달되는 파라미터를 의미합니다. 이는 주로 필터링, 정렬, 페이징 등의 부가 정보를 전달할 때 사용됩니다.

- 특징
  - **쿼리 문자열 기반**: URL의 끝에 `?key=value` 형태로 전달됩니다.
  - **필터링 및 검색**: 여러 필터 조건이나 추가적인 데이터 요청을 전달할 때 사용됩니다.
 
- 주 사용처
  - 데이터 필터링 및 검색 조건 전달: 예를 들어, 사용자의 역할(role)에 따른 필터링, 특정 페이지(page)와 페이지 크기(size)를 전달하여 페이징 처리할 때.
  - 다중 옵션 전달: 예를 들어, 여러 검색 조건이나 정렬 옵션을 전달할 때.
